### DISCUSSION TITLE
is there a way to open the file next to the current buffer ?

### DISCUSSION THREAD

**Discussion by zztczcx** [nvim-tree/nvim-tree.lua#1933] [Q&A]
For example 
if I have 1, 2, 3, 4 buffers. and 2. is focused now. 

when I open a new file from nvim-tree, can I open it next to 2, instead of becoming 5. ?

**Comment by alex-courtis** [comment]
Try playing around with `:help splitright` and `:help splitbelow`.

It would be useful if you showed a diagram of your windows/buffers like https://github.com/nvim-tree/nvim-tree.lua/discussions/1927#discussioncomment-4744178

**Reply by zztczcx** [reply]
what I want is not to split the current buffer. 

I want is 

```
------------------------------------------
|nvim-tree     |  *buffer 1* (current), buffer2, buffer3
------------------------------------------
|    dir       |     file1's content
|       -file1 | 
|       -file2 |
| 
------------------------------------------

```
so  when I open file 2, I won't want it split the window with file 1,
I want it open as normal, but the buffer's position is next to buffer 1., basically like inserting between buffer 1 and 2.

**Reply by alex-courtis** [reply]
>  I want it open as normal, but the buffer's position is next to buffer 1., basically like inserting between buffer 1 and 2.

Confirming: there is only one window visible. You want to open file 2 in that one window. Is that correct?

What do you mean by buffer position? Remember that vim doesn't have any concept of MRU for opened buffers; there is no order other than `#`, the most recently opened.

**Reply by zztczcx** [reply]
>Confirming: there is only one window visible. You want to open file 2 in that one window. Is that correct?

yes, in the same window .

but normally, when you use 'enter' to open file 2,  it will open in buffer 4, which is on buffer 3's right.
 
if file 2  is related to file 1,   I want file2 can open on buffer 1' right side.  when I browser these two files, 
I don't need to use 'tab' to jump over buffer2, and buffer3. 

hope you got what I mean.

**Reply by alex-courtis** [reply]
Are you using a plugin to show "tabs" above windows?

**Reply by zztczcx** [reply]
just bufferline.nvim. that managers buffers.

**Reply by alex-courtis** [reply]
Unfortunately nvim-tree does not interact with bufferline. Perhaps you could build some automation to do this.
